platform/darwin/src/MGLImageSource.h

```diff
@@ -0,0 +1,136 @@
+#import "MGLSource.h"
+
+#import "MGLFoundation.h"
+#import "MGLTypes.h"
+#import "MGLGeometry.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+MGL_EXPORT
+/**
+ `MGLImageSource` is a map content source that supplies a georeferenced raster
+ image to be shown on the map. The geographic location of the raster image content, 
+ supplied with `MGLCoordinateQuad`, can be non-axis aligned.
```

Comment:
Per https://github.com/mapbox/mapbox-gl-native/pull/9110/#discussion_r118759190, we should point out that we have different ways to handle non-georeferenced images:

> An image source is used for a [georeferenced](https://en.wikipedia.org/wiki/Georeferencing) image – that is, an image in which the contents correspond to physical locations on the map. The georeferenced image scales and rotates as the user zooms and rotates the map. Images may also be used as icons or patterns in a style layer. To register an image for use as an icon or pattern, use the `-[MGLStyle setImage:forName:]` method. To configure a point annotation’s image, use the `MGLAnnotationImage` class.